e2e/tests/migration_steps_test.go

```diff
@@ -90,45 +219,381 @@ func (s *migrationSuite) TheShannonServiceConfigIsUpdatedIfApplicable(actorType
+}
+
+// nextMorseUnstakedKeyIdx returns the next morse private key index which is
+// intended to be used for unstaked morse accounts. If the current morseKeyIdx
+// is not an unstaked morse account, morseKeyIdx is incremented until the next
+// Morse key index which should be an unstaked account, given the round-robin
+// distribution of morse account actor types.
+func (s *migrationSuite) nextMorseUnstakedKeyIdx() uint64 {
+	currentIdx := s.getMorseKeyIdx()
+	// Skip non-application account keys.
+	for {
+		if testmigration.GetRoundRobinMorseAccountActorType(currentIdx) ==
+			testmigration.MorseUnstakedActor {
+			break
+		}
+		currentIdx = s.nextMorseKeyIdx()
+	}
+
+	return currentIdx
+}
+
+// nextShannonKeyIdx randomizes the shannon key index and returns a key name
+// which is derived from the new index.
+func (s *migrationSuite) nextShannonKeyIdx() string {
+	shannonKeyIdx = rand.Uint64()
+	return s.getShannonKeyName()
+}
+
+// getShannonKeyName returns the key name derived the current shannon key index.
+func (s *migrationSuite) getShannonKeyName() string {
+	return fmt.Sprintf("shannon-key-%d", shannonKeyIdx)
+}
+
+// getSShannonKeyAddress checks if the key corresponding to the current shannon key index
+// is present in the poktrolld keyring. If it is, it returns the address and true. Otherwise,
+// it returns an empty string and false.
+func (s *migrationSuite) getShannonKeyAddress() (shannonAddr string, isFound bool) {
+	s.buildAddrMap()
+	shannonKeyName := s.getShannonKeyName()
+	shannonAddr, isFound = accNameToAddrMap[shannonKeyName]
+	return shannonAddr, isFound
+}
+
+// queryShowMorseClaimableAccount queries the migration module for the MorseClaimableAccount with the given morseSrcAddress.
+// It will fail the test if the account is not found.
+func (s *migrationSuite) queryShowMorseClaimableAccount(morseSrcAddress string) migrationtypes.MorseClaimableAccount {
+	cmdResult, err := s.pocketd.RunCommandOnHost(
+		"",
+		"query",
+		"migration",
+		"show-morse-claimable-account",
+		morseSrcAddress,
+		"--output=json",
+	)
+	require.NoError(s, err)
+
+	res := new(migrationtypes.QueryMorseClaimableAccountResponse)
+	err = cmtjson.Unmarshal([]byte(cmdResult.Stdout), res)
+	require.NoError(s, err)
+
+	return res.MorseClaimableAccount
+}
+
+// queryListMorseClaimableAccounts queries the migration module for all morse claimable accounts.
+func (s *migrationSuite) queryListMorseClaimableAccounts() []migrationtypes.MorseClaimableAccount {
+	cmdResult, err := s.pocketd.RunCommandOnHost(
+		"",
+		"query",
+		"migration",
+		"list-morse-claimable-account",
+		"--output=json",
+	)
+	require.NoError(s, err)
+
+	res := new(migrationtypes.QueryAllMorseClaimableAccountResponse)
+	err = cmtjson.Unmarshal([]byte(cmdResult.Stdout), res)
+	require.NoError(s, err)
+
+	return res.MorseClaimableAccount
+}
+
+// getCurrentBlockHeight uses poktrolld to query for the current block height.
+func (s *migrationSuite) getCurrentBlockHeight() int64 {
```

Comment:
Why not reuse `getCurrentBlockHeight` from `e2e/tests/init_test.go`?

Reply:
Good call - thanks!